pkg/policy/mapstate.go

```diff
@@ -1633,9 +1645,8 @@ func (mc *MapChanges) consumeMapChanges(p *EndpointPolicy, features policyFeatur
 			p.policyMapState.denyPreferredInsertWithChanges(key, entry, features, changes)
 		} else {
 			// Delete the contribution of this cs to the key and collect incremental changes
-			for cs := range mc.synced[i].Value.owners { // get the sole selector
-				p.policyMapState.deleteKeyWithChanges(mc.synced[i].Key, cs, changes)
-			}
+			cs, _ := mc.synced[i].Value.owners.Get() // get the sole selector
```

Comment:
[Question] Just out of curiosity, why here we'll always have at most one selector?

Reply:
It's due to how `mc` is populated. Multiple owners becomes possible only after entries are added to the MapState.